Pod/Classes/VOKCoreDataManager.m

```diff
@@ -171,47 +147,70 @@ - (void)initPersistentStoreCoordinator
-                              otherButtonTitles:nil] show];
-        }
-        
-        if (self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecoveryAndAlert ||
-            self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecovery) {
-            VOK_CDLog(@"Full database delete and rebuild");
-            [[NSFileManager defaultManager] removeItemAtPath:storeURL.path error:nil];
-            if (![_persistentStoreCoordinator addPersistentStoreWithType:storeType
-                                                           configuration:nil
-                                                                     URL:storeURL
-                                                                 options:nil
-                                                                   error:&error]) {
-                [NSException raise:@"Vokoder Persistant Store Creation Failure after migration"
-                            format:@"Unresolved error %@, %@", error, [error userInfo]];
+    NSError *error;
+    
+    if (![persistentStoreCoordinator addPersistentStoreWithType:storeType
+                                                  configuration:nil
+                                                            URL:storeURL
+                                                        options:options
+                                                          error:&error]) {
+        switch (self.migrationFailureOptions) {
+            case VOKMigrationFailureOptionWipeRecoveryAndAlert:
+            {
+                NSString *title = @"Migration Failed";
+                NSString *message = @"Migration has failed, data will be erased to ensure application stability.";
+                if ([UIAlertController class]) {
+                    UIAlertController *alert = [UIAlertController alertControllerWithTitle:title
+                                                                                   message:message
+                                                                            preferredStyle:UIAlertControllerStyleAlert];
+                    [[UIApplication sharedApplication].keyWindow.rootViewController presentViewController:alert
+                                                                                                 animated:YES
+                                                                                               completion:nil];
+                } else {
+                    //TODO: delete UIAlertView once support is dropped for iOS 7
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+                    [[[UIAlertView alloc] initWithTitle:title
+                                                message:message
+                                               delegate:nil
+                                      cancelButtonTitle:@""
+                                      otherButtonTitles:nil] show];
+#pragma clang diagnostic pop
```

Comment:
Worth wrapping all this in an `if iOS 7 or lower` block and using UIAlertController otherwise?